docs/release-notes.md

```diff
@@ -1,3 +1,31 @@
+## 1.0.0
+
+Wow!!! 1.0.0 is here! 🎉
+
+### Removed
+
+* Removed `WSGIMiddleware`, which is deprecated since `0.19.0`. Please use [`a2wsgi`](https://github.com/abersheeran/a2wsgi) instead.
+* Removed `run_until_first_complete`, which is deprecated since `0.19.0`.
+* Removed `iscoroutinefunction_or_partial`, which is deprecated since `0.20.1`.
+  It was an internal function, which we have replaced by `_utils.is_async_callable`.
+* Removed `WS_1004_NO_STATUS_RCVD` and `WS_1005_ABNORMAL_CLOSURE` from the `status` module, which were deprecated since `0.19.1`.
+* Removed `ExceptionMiddleware` from the `exceptions` module, which was deprecated since `0.19.1`.
+  The same middleware can be found in the `middleware.exceptions` module.
+
+### Deprecated
+
+* Deprecated `BaseHTTPMiddleware`. Please refer to [Pure ASGI Middleware](https://www.starlette.io/middleware/#pure-asgi-middleware)
+  for recommended approach.
+* Deprecated `Router.on_event` decorator. Please refer to [Registering events](https://www.starlette.io/events/#registering-events)
+  for recommended approach.
+* Deprecated `Router.websocket_route` decorator. Please refer to [WebSocket Routing](https://www.starlette.io/routing/#websocket-routing)
+  for recommended approach.
+* Deprecated `Router.route` decorator. Please refer to [HTTP Routing](https://www.starlette.io/routing/#http-routing) for recommended approach.
+* Deprecated `Router.host` method. Please refer to [Host-based routing](https://www.starlette.io/routing/#host-based-routing)
+  for recommended approach.
+* Deprecated `Router.mount` method. Please refer to [Submounting routes](https://www.starlette.io/routing/#submounting-routes)
+  for recommended approach.
+
```

Comment:
For FastAPI, at some point, I plan on deprecating some things, and then removing them, all that before the first 1.0.0 release.

I would like the first 1.0.0 release in FastAPI to not have sort of dangling or pending things to do. It's also easier to deprecate, remove, revert, add, and change things in 0.x.x land. Currently removing deprecated things requires a minor bump (the same as in Starlette right now). But removing things can break others and require updates or reverts. I want to have those things a bit more stable and not expecting changes before releasing the first FastAPI 1.0.0.

But then again, that's just how I plan on doing it on FastAPI, it doesn't have to be that way, there's no obvious single right way to do things. But knowing that I plan on doing that in FastAPI might relax the requirements to release a first 1.0.0 here before deprecating and removing those things.

Also, I have Starlette strictly pinned in FastAPI, so if these changes were done in 0.x.x versions they wouldn't affect FastAPI users, as the internal parts would be updated before the version pin is raised.

I don't have strong opinions about how it should be done here, just my random thoughts.

Reply:
https://github.com/encode/starlette/pull/1888#issuecomment-1267325021 :+1: